lgc/patch/PatchBufferOp.cpp

```diff
@@ -420,12 +420,39 @@ void PatchBufferOp::visitCallInst(CallInst &callInst) {
   } else if (callName.startswith(lgcName::LateBufferLength)) {
     Value *const pointer = getPointerOperandAsInst(callInst.getArgOperand(0));
 
-    // Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
-    Value *const bufferLength = m_builder->CreateExtractElement(m_replacementMap[pointer].first, 2);
+    // Compute the length of the variable sized array
+    // 1. Extract element 2 which is the NUM_RECORDS field from the buffer descriptor.
+    Value *const bufferDesc = m_replacementMap[pointer].first;
+    Value *numRecords = m_builder->CreateExtractElement(bufferDesc, 2);
+
+    Value *offset = callInst.getArgOperand(1);
+    if (cast<ConstantInt>(offset)->getZExtValue() > 0) {
```

Comment:
You don't need to make this conditional. IRBuilder (or a later pass that does simplification) will spot that you're subtracting constant 0, and will remove the sub. Also, if you implement my other comment that the Builder API should take a Value, you can't assume here that it is a ConstantInt.

Reply:
Removed the conditional.